Client/Frontend/Settings/SettingsContentViewController.swift

```diff
@@ -0,0 +1,180 @@
+    var url: NSURL!
+    var debug: Bool = false
+    private var timer: NSTimer?
+
+    private var isLoaded: Bool = false {
+        didSet {
+            if isLoaded {
+                UIView.transitionFromView(interstitialView, toView: webView,
+                    duration: 0.5,
+                    options: UIViewAnimationOptions.TransitionCrossDissolve,
+                    completion: { finished in
+                        self.interstitialView.removeFromSuperview()
+                        self.interstitialSpinnerView.stopAnimating()
+                })
+            }
+        }
+    }
+
+    private var isError: Bool = false {
+        didSet {
+            if isError {
+                interstitialErrorView.hidden = false
+                UIView.transitionFromView(interstitialSpinnerView, toView: interstitialErrorView,
+                    duration: 0.5,
+                    options: UIViewAnimationOptions.TransitionCrossDissolve,
+                    completion: { finished in
+                        self.interstitialSpinnerView.removeFromSuperview()
+                        self.interstitialSpinnerView.stopAnimating()
+                })
+            }
+        }
+    }
+
+    // The view shown while the content is loading in the background web view.
+    private var interstitialView: UIView!
+    private var interstitialSpinnerView: UIActivityIndicatorView!
+    private var interstitialErrorView: UILabel!
+
+    // The web view that displays content from the fxa-content-server.
+    private var webView: WKWebView!
+
+    func startLoading(timeout: Double = DefaultTimeoutTimeInterval) {
+        if self.isLoaded {
+            return
+        }
+        if timeout > 0 {
+            self.timer = NSTimer.scheduledTimerWithTimeInterval(timeout, target: self, selector: "SELdidTimeOut", userInfo: nil, repeats: false)
+        } else {
+            self.timer = nil
+        }
+        self.webView.loadRequest(NSURLRequest(URL: url))
+        self.interstitialSpinnerView.startAnimating()
+    }
+
+    private func layoutInnerViews() {
+        self.webView.snp_remakeConstraints { make in
+            make.left.right.top.equalTo(self.view)
+            make.bottom.equalTo(self.view)
+        }
+        if !isLoaded {
+            // If the page has loaded, the interstitial view will have been removed.
+            self.interstitialView.snp_remakeConstraints { make in
+                make.left.right.top.equalTo(self.view)
+                make.bottom.equalTo(self.view)
+            }
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Don't extend under navigation bar.
+        self.edgesForExtendedLayout = UIRectEdge.None
+
+        // This background agrees with the content server background.
+        // Keeping the background constant prevents a pop of mismatched color.
+        view.backgroundColor = UIColor(red: 242 / 255.0, green: 242 / 255.0, blue: 242 / 255.0, alpha: 1.0)
```

Comment:
What this does is keep the background constant so that weird things don't pop up during the interstitial (i think). I confirmed by taking out these comments and random highlights from the page behind the interstitial would flash before the interstitial was done animating.